src/webapp/hooks/useAppConfig.ts

```diff
@@ -0,0 +1,60 @@
+import { useAppContext } from "../contexts/app-context";
+import { defaultConfig, PersistedConfig } from "../../data/entities/PersistedConfig";
+import React, { useMemo, useState } from "react";
+import { getCustomizableAppText } from "../../domain/entities/CustomText";
+import _ from "lodash";
+
+export function useAppConfig() {
+    const { usecases } = useAppContext();
+    const [appConfig, setAppConfig] = React.useState<PersistedConfig>(defaultConfig);
+    const [hasSettingsAccess, setHasSettingsAccess] = useState(false);
+    const [hasLoaded, setHasLoaded] = useState(false);
+    const appCustomText = useMemo(() => getCustomizableAppText(appConfig?.customText || {}), [appConfig]);
+    const logoInfo = useMemo(() => getLogoInfo(appConfig?.logo), [appConfig]);
+
+    const save = React.useCallback(
+        async (config: PersistedConfig) => {
+            return usecases.config.save(config).then(setAppConfig);
+        },
+        [usecases.config]
+    );
+
+    React.useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const config = await usecases.config.get();
+                setAppConfig(config);
+                const hasAccess = await usecases.user.checkSettingsPermissions(config);
+                setHasSettingsAccess(hasAccess);
+            } catch (error) {
+                console.error("Error:", error);
+            } finally {
+                setHasLoaded(true);
+            }
+        };
+
+        fetchData();
```

Comment:
Should we use `await`?

Reply:
No, the function passed to useEffect cannot be async, that's why we define the wrapper.